spec/requests/api/tenant_quotas_spec.rb

```diff
@@ -0,0 +1,263 @@
+
+      tenant = FactoryGirl.create(:tenant, :parent => root_tenant)
+      quota = FactoryGirl.create(:tenant_quota, :tenant_id => tenant.id, :name => :cpu_allocated, :value => 1)
+
+      options = {:value => 5}
+
+      run_post "/api/tenants/#{tenant.id}/quotas/#{quota.id}", gen_request(:edit, options)
+
+      expect_request_success
+      quota.reload
+      expect(quota.value).to eq(5)
+    end
+
+    it "can update a quota from a tenant with PUT" do
+      api_basic_authorize action_identifier(:quotas, :edit, :subresource_actions, :put)
+
+      tenant = FactoryGirl.create(:tenant, :parent => root_tenant)
+      quota = FactoryGirl.create(:tenant_quota, :tenant_id => tenant.id, :name => :cpu_allocated, :value => 1)
+
+      options = {:value => 5}
+
+      run_put "/api/tenants/#{tenant.id}/quotas/#{quota.id}", options
+
+      expect_request_success
+      quota.reload
+      expect(quota.value).to eq(5)
+    end
+
+    it "can update multiple quotas from a tenant with POST" do
+      api_basic_authorize action_identifier(:quotas, :edit, :subcollection_actions, :post)
+      tenant = FactoryGirl.create(:tenant, :parent => root_tenant)
+      quota_1 = FactoryGirl.create(:tenant_quota, :tenant_id => tenant.id, :name => :cpu_allocated, :value => 1)
+      quota_2 = FactoryGirl.create(:tenant_quota, :tenant_id => tenant.id, :name => :mem_allocated, :value => 2)
+
+      options = [
+        {"href" => "/api/tenants/#{tenant.id}/quotas/#{quota_1.id}", "value" => 3},
+        {"href" => "/api/tenants/#{tenant.id}/quotas/#{quota_2.id}", "value" => 4},
+      ]
+
+      run_post "/api/tenants/#{tenant.id}/quotas/", gen_request(:edit, options)
+
+      expect_request_success
+      expect_results_to_match_hash(
+        "results",
+        [{"id" => quota_1.id, "value" => 3},
+         {"id" => quota_2.id, "value" => 4}]
+      )
+      expect(quota_1.reload.value).to eq(3)
+      expect(quota_2.reload.value).to eq(4)
+    end
+
+    it "can delete a quota from a tenant with POST" do
+      api_basic_authorize action_identifier(:quotas, :delete, :subresource_actions, :post)
+
+      tenant = FactoryGirl.create(:tenant, :parent => root_tenant)
+      quota = FactoryGirl.create(:tenant_quota, :tenant_id => tenant.id, :name => :cpu_allocated, :value => 1)
+
+      expect do
+        run_post "/api/tenants/#{tenant.id}/quotas/#{quota.id}", gen_request(:delete)
+      end.to change(TenantQuota, :count).by(-1)
+
+      expect_request_success
+    end
+
+    it "can delete a quota from a tenant with DELETE" do
+      api_basic_authorize action_identifier(:quotas, :delete, :subresource_actions, :delete)
+
+      tenant = FactoryGirl.create(:tenant, :parent => root_tenant)
+      quota = FactoryGirl.create(:tenant_quota, :tenant_id => tenant.id, :name => :cpu_allocated, :value => 1)
+
+      expect do
+        run_delete "/api/tenants/#{tenant.id}/quotas/#{quota.id}", gen_request(:delete)
```

Comment:
I think we don't need to specify the action (`gen_request(:delete)`) if we're using the DELETE verb